src/azure-cli-core/azure/cli/core/prompting.py

```diff
@@ -22,28 +23,64 @@ def _verify_is_a_tty():
         raise NoTTYException()
 
 
-def prompt(msg):
+def prompt(msg, help_string=None):
     _verify_is_a_tty()
-    return input(msg)
+    while True:
+        val = input(msg)
+        if val == '?' and help_string is not None:
+            print(help_string)
+            continue
+        return val
 
 
-def prompt_pass(msg='Password: ', confirm=False):
+def prompt_int(msg, help_string=None):
     _verify_is_a_tty()
-    password = getpass.getpass(msg)
-    if confirm:
-        password2 = getpass.getpass('Confirm ' + msg)
-        assert password == password2, 'Passwords do not match.'
-    return password
+
+    while True:
+        value = input(msg)
+        if value == '?' and help_string is not None:
+            print(help_string)
+            continue
+        try:
+            return int(value)
+        except ValueError:
+            logger.warning('%s is not a valid number', value)
+
+
+def prompt_pass(msg='Password: ', confirm=False, help_string=None):
+    _verify_is_a_tty()
+    while True:
+        password = getpass.getpass(msg)
+        if password == '?' and help_string is not None:
+            print(help_string)
+            continue
+        if confirm:
+            password2 = getpass.getpass('Confirm ' + msg)
+            if password != password2:
+                logger.warning('Passwords do not match.')
+                continue
+        return password
+
+
+def prompt_y_n(msg, default=None, help_string=None):
+    return _prompt_bool(msg, 'y', 'n', default=default, help_string=help_string)
+
+
+def prompt_t_f(msg, default=None, help_string=None):
+    return _prompt_bool(msg, 't', 'f', default=default, help_string=help_string)
 
 
-def prompt_y_n(msg, default=None):
+def _prompt_bool(msg, true_str, false_str, default=None, help_string=None):
```

Comment:
Prompting "yes" or "no" essentially is just a particular case prompting multiple choices. In my opinion we can combine these two methods. However, I do not intend to block this PR since the change as it works fine for now.